lib/plugins/throttle.js

```diff
@@ -278,7 +280,16 @@ function throttle(options) {
         req.log.trace('Throttle(%s): num_tokens= %d',
             attr, bucket.tokens);
 
-        if (!bucket.consume(1)) {
+        var tooManyRequests = !bucket.consume(1);
+
+        // set throttle headers after consume which changes the remaining tokens
+        if (options.setHeaders) {
```

Comment:
At the top of this function, we should assert `options.setHeaders` is an optional boolean.